src/gcp/runtimeconfig.ts

```diff
@@ -50,7 +50,7 @@ function deleteConfig(projectId: string, configId: string): Promise<any> {
     });
 }
 
-function listVariables(configPath: string): Promise<any> {
+function listVariables(configPath: string): Promise<{ name: string }[] | undefined> {
```

Comment:
what do you think that if instead of returning either an arry or an undefined, we always returned an array by doing something like this:

```
...
    .then((resp) => {
      return Promise.resolve(resp.body.variables || []);
    });
```

Then we won't need any of the `if (!variables) return {};` jazz.